src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -305,9 +305,17 @@ void ImageReplayer<I>::start(Context *on_finish, bool manual, bool restart)
       dout(10) << "canceled restart" << dendl;
       r = -ECANCELED;
     } else {
-      m_state = STATE_STARTING;
+      // checking if description is set to "orphan (force promote)"
+      // in which case we want to preserve the status.
+      if (m_state == STATE_STOPPED &&
+          m_state_desc.find("orphan") != std::string::npos) {
+        dout(10) << "canceling start" << dendl;
+        r = -ECANCELED;
```

Comment:
This I'm not following at all.  IIUC this just prevents the replayer from being started (ever again until rbd-mirror daemon is killed and respawned?).  If the image gets force promoted with the opposite cluster being available, after the image is demoted and resynced there in order to reestablish mirroring, this would lead to the image showing up as "unknown" and health going to WARNING there.

Reply:
Lets target the preserve part as part of a separate PR as discussed